### PR TITLE
RFC: Login docs

### PR DESCRIPTION
**Pull request by provokateurin**
https://github.com/provokateurin/nextcloud-neon/issues/343

### CODE REVIEW

**Comment by Leptopoda** [comment]
Looks good.
I'd like to make the validations (validate server details) and (validate login credentials) a condition. The validation can fail and what happens.
Or do you think this is irrelevant for this chart?

Does it make sense to name the end state? Like logged in?

---

a more general question:
I think once our doc comments are mature enough we probably want to host the generated docs either here on GH-Pages or by pub.dev once published. In this case we'd probably want to also publish this with it. I've never used it but dart doc supports topics. Arbitrary markdown that will be added to the generated docs. I only saw that in the [go_router docs](https://pub.dev/documentation/go_router/latest/) and think this could be good to use.

In that case we should start with a structure that'll support this flow (separate md per documentation part). 

---

How didn't I know plantuml before. I've only used umlet and it's just awefull. Would've made a lot easier.

File: docs/login.puml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    endif
+endif
+
+if "Selected login method?" then
+-[#blue]->[Login flow] "Validate server details"
+-[#blue]-> "Open login flow page"
+-[#blue]-> "Initiate login flow"
+-[#blue]-> "Open web browser"
+-[#blue]-> "Wait for login flow result"
+-[#blue]-> "Validate login details"
```
Comment (provokateurin) [inline docs/login.puml:19]: This one is obsolete, as we can be sure that the credentials returned by the login flow are correct (nevertheless we need to fetch the username, so we might as well leave it in)

Comment (provokateurin) [reply docs/login.puml:19]: @Leptopoda Should we leave it or not? The logic would be exactly the same and we should also display some UI

**Comment by provokateurin** [comment]
I tried adding the failure cases but the diagram looks horrible :/

**Comment by provokateurin** [comment]
And the diagram gets way too complicated because it's not possible to properly reuse these nodes.

**Review by Leptopoda — APPROVED** [review]

**Pull request merged**